wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureConversationGuestLinkViewModel.swift

```diff
@@ -18,26 +18,40 @@
 
 import Foundation
 import UIKit
+import WireSyncEngine
 
 // MARK: - CreatePasswordSecuredLinkViewModelDelegate
 
 // sourcery: AutoMockable
 protocol CreatePasswordSecuredLinkViewModelDelegate: AnyObject {
-    func viewModel(_ viewModel: CreateSecureGuestLinkViewModel, didGeneratePassword password: String)
+    func viewModel(_ viewModel: CreateSecureConversationGuestLinkViewModel, didGeneratePassword password: String)
+    func viewModelDidValidatePasswordSuccessfully(_ viewModel: CreateSecureConversationGuestLinkViewModel)
+    func viewModel(_ viewModel: CreateSecureConversationGuestLinkViewModel, didFailToValidatePasswordWithReason reason: String)
+    func viewModel(_ viewModel: CreateSecureConversationGuestLinkViewModel, didCreateLink link: String)
+    func viewModel(_ viewModel: CreateSecureConversationGuestLinkViewModel, didFailToCreateLinkWithError error: Error)
 }
 
 // MARK: - CreateSecureGuestLinkViewModel
 
-final class CreateSecureGuestLinkViewModel {
+final class CreateSecureConversationGuestLinkViewModel {
+
+    enum LinkCreationError: Error {
+        case underfinedLink
```

Comment:
What does this mean?

Reply:
I needed to pass an error when success is nil:

```swift
conversationGuestLinkUseCase.invoke(conversation: conversation, password: password) { [weak self] result in
            guard let self = self else { return }

            switch result {
            case .success(let link?):
                self.delegate?.viewModel(self, didCreateLink: link)
            case .success(nil):
                self.delegate?.viewModel(self, didFailToCreateLinkWithError: LinkCreationError.underfinedLink)
            case .failure(let error):
                self.delegate?.viewModel(self, didFailToCreateLinkWithError: error)
            }
        }
```

Reply:
Did you find out from backend when we can expect to have a nil link?